cristin-person/src/main/java/no/unit/nva/cristin/person/model/nva/NvaIdentifier.java

```diff
@@ -0,0 +1,125 @@
+    public String getType() {
+        return type;
+    }
+
+    public String getValue() {
+        return value;
+    }
+
+    @JacocoGenerated
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof NvaIdentifier)) {
+            return false;
+        }
+        NvaIdentifier that = (NvaIdentifier) o;
+        return Objects.equals(getType(), that.getType()) && Objects.equals(getValue(), that.getValue());
+    }
+
+    @JacocoGenerated
+    @Override
+    public int hashCode() {
+        return Objects.hash(getType(), getValue());
+    }
+
+    @JacocoGenerated
+    public static final class Builder {
+
+        private transient String type;
+        private transient String value;
+
+        public Builder withType(String type) {
+            this.type = type;
+            return this;
+        }
+
+        public Builder withValue(String value) {
+            this.value = value;
+            return this;
+        }
+
+        public NvaIdentifier build() {
+            return new NvaIdentifier(this.type, this.value);
+        }
+    }
+
+    /**
+     * Transforms identifiers from Cristin into a List.
+     *
+     * @param cristinPerson Cristin model containing identifiers.
+     * @return List of transformed identifiers
+     */
+    public static List<NvaIdentifier> identifiersFromCristinPerson(CristinPerson cristinPerson) {
+        List<NvaIdentifier> identifiers = new ArrayList<>();
+
+        Optional.of(cristinPerson)
+            .map(CristinPerson::getCristinPersonId)
+            .ifPresent(id -> identifiers.add(createOneIdentifier("CristinIdentifier", id)));
+
+        Optional.of(cristinPerson)
+            .map(CristinPerson::getOrcid)
+            .map(CristinOrcid::getId)
+            .ifPresent(id -> identifiers.add(createOneIdentifier("ORCID", id)));
+
+        return identifiers;
+    }
+
+    /**
+     * Transforms names from a Cristin person into a List.
+     *
+     * @param cristinPerson Cristin model containing name identifiers.
+     * @return List of transformed names
+     */
+    public static List<NvaIdentifier> namesFromCristinPerson(CristinPerson cristinPerson) {
+        List<NvaIdentifier> names = new ArrayList<>();
+
+        Optional.of(cristinPerson)
+            .map(CristinPerson::getFirstName)
+            .ifPresent(id -> names.add(createOneIdentifier("FirstName", id)));
+
+        Optional.of(cristinPerson)
+            .map(CristinPerson::getSurname)
+            .ifPresent(id -> names.add(createOneIdentifier("LastName", id)));
+
+        Optional.of(cristinPerson)
+            .map(CristinPerson::getFirstNamePreferred)
+            .ifPresent(id -> names.add(createOneIdentifier("PreferredFirstName", id)));
+
+        Optional.of(cristinPerson)
+            .map(CristinPerson::getSurnamePreferred)
+            .ifPresent(id -> names.add(createOneIdentifier("PreferredLastName", id)));
+
+        return names;
```

Comment:
It seems that these are the names of the fields of the CirstinPerson. This code belongs to CristinPerson, basically I agree with Rurik that the "fromX" methods should be "toX" methods. But in any case it is better to use constants instead of magic strings.